crates/bevy_reflect/src/type_uuid.rs

```diff
@@ -28,3 +28,49 @@ where
         std::any::type_name::<Self>()
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate as bevy_reflect;
+    use bevy_reflect_derive::TypeUuid;
+
+    #[derive(TypeUuid)]
+    #[uuid = "af6466c2-a9f4-11eb-bcbc-0242ac130002"]
+    struct TestDeriveStruct<T>
+    where
+        T: Clone,
+    {
+        _value: T,
+    }
+
+    fn test_impl_type_uuid(_: &impl TypeUuid) {}
+
+    #[test]
+    fn test_generic_type_uuid_derive() {
+        #[derive(TypeUuid, Clone)]
+        #[uuid = "ebb16cc9-4d5a-453c-aa8c-c72bd8ec83a2"]
+        struct T;
+
+        let test_struct = TestDeriveStruct { _value: T };
+        test_impl_type_uuid(&test_struct);
+    }
+
+    #[test]
+    fn test_generic_type_unique_uuid() {
```

Comment:
I'd like to see a variant on this test to verify that the behavior is as expected when `A` and `B` have the same UUID.

(I'm pretty sure that this would and should result in the compound UUID being the same)

Reply:
Done. (and yes, the UUIDs are the same)